corehq/apps/api/resources/v0_5.py

```diff
@@ -1091,13 +1091,28 @@ def _process_params(self, domain, params):
                 val = params.getlist(param)
             processed_params[param] = val
 
+            if param == 'limit':
+                processed_params['limit'] = self._process_limit(val)
+
         if 'local_timezone' in processed_params:
             self.local_timezone = pytz.timezone(processed_params['local_timezone'])
         else:
             self.local_timezone = Domain.get_by_name(domain).get_default_timezone()
 
         return processed_params
 
+    def _process_limit(self, limit):
```

Comment:
does tastypie not do this already?  Or is that only if you use the default methods?

Reply:
It does do error checking for Negative values, but not before `cursor_query` is called and causing `(queryset[:int(params['limit'])])` to error first.

And if limit is > max_limit, tastypie doesn't raise an error but just reduces the number of objects in the response.